rnd/autogpt_builder/src/components/Flow.tsx

```diff
@@ -153,60 +163,98 @@ const Flow: React.FC = () => {
+      return {};
+    }
 
-    if (schema.properties) {
-      Object.keys(schema.properties).forEach((key) => {
-        if (values[key] !== undefined) {
-          if (schema.properties[key].type === 'object') {
-            inputData[key] = getNestedData(schema.properties[key], values[key]);
-          } else {
-            inputData[key] = values[key];
+    const getNestedData = (schema: ObjectSchema, values: { [key: string]: any }): { [key: string]: any } => {
+      let inputData: { [key: string]: any } = {};
+
+      if (schema.properties) {
+        Object.keys(schema.properties).forEach((key) => {
+          if (values[key] !== undefined) {
+            if (schema.properties[key].type === 'object') {
+              inputData[key] = getNestedData(schema.properties[key], values[key]);
+            } else {
+              inputData[key] = values[key];
+            }
           }
-        }
-      });
-    }
+        });
+      }
 
-    if (schema.additionalProperties) {
-      inputData = { ...inputData, ...values };
-    }
+      if (schema.additionalProperties) {
+        inputData = { ...inputData, ...values };
+      }
 
-    return inputData;
-  };
+      return inputData;
+    };
 
-  let inputData = getNestedData(blockSchema, node.data.hardcodedValues);
+    let inputData = getNestedData(blockSchema, node.data.hardcodedValues);
 
-  // Get data from connected nodes
-  const incomingEdges = allEdges.filter(edge => edge.target === node.id);
-  incomingEdges.forEach(edge => {
-    const sourceNode = allNodes.find(n => n.id === edge.source);
-    if (sourceNode && sourceNode.data.output_data) {
-      const outputKey = Object.keys(sourceNode.data.output_data)[0]; // Assuming single output
-      inputData[edge.targetHandle as string] = sourceNode.data.output_data[outputKey];
-    }
-  });
+    // Get data from connected nodes
+    const incomingEdges = allEdges.filter(edge => edge.target === node.id);
+    incomingEdges.forEach(edge => {
+      const sourceNode = allNodes.find(n => n.id === edge.source);
+      if (sourceNode && sourceNode.data.output_data) {
+        const outputKey = Object.keys(sourceNode.data.output_data)[0]; // Assuming single output
+        inputData[edge.targetHandle as string] = sourceNode.data.output_data[outputKey];
+      }
+    });
 
-  // Filter out any inputs that are not in the block's schema
-  Object.keys(inputData).forEach(key => {
-    if (!blockSchema.properties[key]) {
-      delete inputData[key];
-    }
-  });
+    // Filter out any inputs that are not in the block's schema
+    Object.keys(inputData).forEach(key => {
+      if (!blockSchema.properties[key]) {
+        delete inputData[key];
+      }
+    });
 
-  console.log(`Final prepared input for ${node.data.blockType} (${node.id}):`, inputData);
-  return inputData;
-};
+    console.log(`Final prepared input for ${node.data.blockType} (${node.id}):`, inputData);
+    return inputData;
+  };
```

Comment:
this is a very strange diff render of a simple indentation fix

Reply:
I can revert these changes if it helps to get this merged

Reply:
nah its okay, we should get the indentation fixes in, itll become more frustrating down the line